controllers/storagecluster/storageclasses.go

```diff
@@ -232,6 +234,16 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 					r.Log.Info("Failed to create StorageClass.", "StorageClass", klog.KRef(sc.Namespace, sc.Name))
 					return err
 				}
+				scRecreated = true
+			}
+			if !scRecreated {
```

Comment:
I would recreate the SC in line 232 with the annotations and labels already present. The labels are also important.

If the reconcile fails to update the SC annotations in this line, a future reconcile will not have the copy of the existing annotations etc. and hence this will never be set on the recreated SC.

Reply:
The SC being recreated on line 232 already has the required annotations and labels if any. 

If it is not recreated (same params), only then we try to add the annotation by updating the SC.

This is what you meant, right?

Reply:
If SC is not recreated you are updating annotations to `existing` SC and updating it, this is fine.

I am still missing how an SC that is recreated (due to parameters differing) preserves annotations and labels from `existing`? IOW this [comment](https://github.com/red-hat-storage/ocs-operator/pull/2817#discussion_r1829111436)

Reply:
We need to preserve the annotation/labels from the SC before delete and reapply that before creating.

Reply:
I would say doing so is out of scope of this PR. The SC was being recreated earlier the same way. 

As far as key rotation is concerned, we consider storage cluster as source of truth. We add/remove annotations to SC accordingly. WDYT?

Reply:
Sure, let's open a tracker for it.

Reply:
Perfect! I'd gladly follow this up as an enhancement. Any suggestions specific to KR on this PR?

Reply:
https://github.com/red-hat-storage/ocs-operator/blob/99fdb662050ddbd412e0a240c477e84ea9dab25d/config/rbac/role.yaml#L312-L321 RBAC comment i mentioned below need to be addressed, we dont have update access

Reply:
@Madhu-1 done!